radix-engine/src/blueprints/account/blueprint.rs

```diff
@@ -61,18 +62,500 @@ impl SecurifiedRoleAssignment for SecurifiedAccount {
 
 impl PresecurifiedRoleAssignment for SecurifiedAccount {}
 
-pub const ACCOUNT_VAULT_COLLECTION_INDEX: CollectionIndex = 0u8;
-pub type AccountVaultEntryContents = Vault;
-
-pub const ACCOUNT_RESOURCE_PREFERENCE_COLLECTION_INDEX: CollectionIndex = 1u8;
-pub type AccountResourcePreferenceEntryContents = ResourcePreference;
+declare_native_blueprint_state! {
+    blueprint_ident: Account,
+    blueprint_snake_case: account,
+    features: {
+    },
+    fields: {
+        account:  {
+            ident: Account,
+            field_type: {
+                kind: StaticSingleVersioned,
+            },
+            condition: Condition::Always,
+        }
+    },
+    collections: {
+        resource_vaults: KeyValue {
+            entry_ident: ResourceVault,
+            key_type: {
+                kind: Static,
+                content_type: ResourceAddress,
+            },
+            value_type: {
+                kind: StaticSingleVersioned,
+            },
+            allow_ownership: true,
+        },
+        resource_preferences: KeyValue {
+            entry_ident: ResourcePreference,
+            key_type: {
+                kind: Static,
+                content_type: ResourceAddress,
+            },
+            value_type: {
+                kind: StaticSingleVersioned,
+            },
+            allow_ownership: false,
+        },
+        authorized_depositors: KeyValue {
+            entry_ident: AuthorizedDepositor,
+            key_type: {
+                kind: Static,
+                content_type: ResourceOrNonFungible,
+            },
+            value_type: {
+                kind: StaticSingleVersioned,
+            },
+            allow_ownership: false,
+        },
+    }
+}
 
-pub const ACCOUNT_AUTHORIZED_DEPOSITORS_COLLECTION_INDEX: CollectionIndex = 2u8;
-pub type AccountAuthorizedDepositorEntryContents = ();
+pub type AccountAccountV1 = AccountSubstate;
```

Comment:
Can have a different name than AccountAccount?

Also, we need to define the type alias before the symbol is referenced in the macro?